app.js

```diff
@@ -46,40 +46,65 @@ app.use("/", indexRouter);
 app.use("/users", usersRouter);
 
 io.sockets.on("connection", function (socket, client) {
-  console.log("client connected!")
-  socket.on('room', function(room){
+  console.log("client connected!");
+  socket.on("room", function (room) {
     socket.join(room);
-    console.log(`Someone joined the room: ${room}`)
-  })
+    console.log(`Someone joined the room: ${room}`);
+  });
   socket.on("newMessage", async (data) => {
     try {
-        console.log(data);
-        Topic.findById(data.topicId, async function(err,Topic){
-          if(err){
-            throw new Error (err);
+      console.log(data);
+      Topic.findById(data.topicId, async function (err, Topic) {
+        if (err) {
+          throw new Error(err);
+        }
+        let user = await User.findById(data.userId, function (err, user) {
+          if (err) {
+            throw new Error(err);
           }
-          let user = await User.findById(data.userId, function(err,user){
-            if (err){
-              throw new Error(err)
-            }
-            return user.username;
-          })
-          console.log(user.username);
-          let newChat = {
-            username: user.username,
-            replyTo: data.replyTo,
-            description: data.description,
-            userId: user._id,
-          }
-          Topic.chats.push(newChat)
-          await Topic.save();
-          io.to(data.topicId).emit('newChat', Topic.chats.pop())
-        })
-
-    } catch{
-        console.log('Error')
+          return user.username;
+        });
+        console.log(user.username);
+        let newChat = {
```

Comment:
Encapsulating this in a separate model (say Message) would make it easier to reference it for Announcement and Tasks creation